tests/components/smarla/conftest.py

```diff
@@ -22,23 +22,24 @@ def mock_config_entry() -> MockConfigEntry:
         domain=DOMAIN,
         unique_id=MOCK_SERIAL_NUMBER,
         source=SOURCE_USER,
+        data=MOCK_USER_INPUT,
     )
 
 
 @pytest.fixture
-def mock_refresh_token_success():
-    """Patch Connection.refresh_token to return True."""
-    with patch.object(Connection, "refresh_token", new=AsyncMock(return_value=True)):
-        yield
-
-
-@pytest.fixture
-def malformed_token_patch():
-    """Patch Connection to raise exception."""
-    return patch.object(Connection, "__init__", side_effect=ValueError)
-
-
-@pytest.fixture
-def invalid_auth_patch():
-    """Patch Connection.refresh_token to return False."""
-    return patch.object(Connection, "refresh_token", new=AsyncMock(return_value=False))
+def mock_connection():
+    """Patch Connection object."""
+    with (
+        patch(
+            "homeassistant.components.smarla.config_flow.Connection"
+        ) as mock_connection,
```

Comment:
```suggestion
    with (
        patch(
            "homeassistant.components.smarla.config_flow.Connection", autospec=True
        ) as mock_connection,
```
This will give your mock knowledge about the class